### ISSUE TITLE
InvalidArgumentError in tensorflow reduce_sum gradient when compiling from sources (still a problem in TF 1.7)

### ISSUE THREAD

**Issue by hccheng** [tensorflow/tensorflow#18353]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 10
- **TensorFlow installed from (source or binary)**: source, without AVX support
- **TensorFlow version (use command below)**: 1.7.0
- **Python version**: 3.5.4
- **Bazel version (if compiling from source)**: CMake 3.10.3
- **GCC/Compiler version (if compiling from source)**: MSVC 2017
- **CUDA/cuDNN version**: CPU only
- **GPU model and memory**: CPU only
- **Exact command to reproduce**:
Using the same example provided in [#12177](https://github.com/tensorflow/tensorflow/issues/12177) and following the suggestions in the comment:
```
import tensorflow as tf
import numpy as np

x = tf.placeholder(tf.float32, [2,3])
y = tf.placeholder(tf.float32, [2,3])
W = tf.Variable(tf.random_normal([3,3], stddev=0.01))

z = tf.matmul(x, W)
# works when removing the reduce_sum op
z = tf.reduce_sum(tf.multiply(z, y), axis=1, keep_dims=True)

optimizer = tf.train.AdamOptimizer()
train_op = optimizer.minimize(z)

Xdata = np.array([[1,2,3],[4,5,6]])
Ydata = np.array([[2,3,4],[3,4,5]])

op = tf.get_default_graph().get_operation_by_name("gradients/Sum_grad/Reshape")
with tf.Session() as sess:
    for input_ in op.inputs:
        print(input_, sess.run(input_, feed_dict={x: Xdata, y: Ydata}))
    sess.run(tf.global_variables_initializer())
    sess.run(train_op, feed_dict={ x: Xdata, y: Ydata })
    print('done')
```
Results:
- Build from source
```
Tensor("gradients/Fill:0", shape=(2, 1), dtype=float32) [[1.]
 [1.]]
Tensor("gradients/Sum_grad/DynamicStitch:0", shape=(?,), dtype=int32) [2 1 0 ... 0 0 0]
2018-04-09 14:59:09.801431: F C:\Users\hcheng\tensorflow\tensorflow\core\framework\tensor_shape.cc:243] Check failed: ndims_byte() < MaxDimensions() (unsigned char value 254 vs. 254)Too many dimensions in tensor
```
- Precompiled binary
```
Tensor("gradients/Fill:0", shape=(2, 1), dtype=float32) [[1.]
 [1.]]
Tensor("gradients/Sum_grad/DynamicStitch:0", shape=(?,), dtype=int32) [2 1]
done
```

**Comment by mrry** [comment]
Well, it looks like something is trying to make a shape of rank `0 - 2`. Since our binaries don't crash in the same way, we'll need some help to track down the problem. Can you try running the script with the environment variable `TF_CPP_MIN_VLOG_LEVEL=1` set, and post the log that results? That should tell us what op failed.

**Comment by hccheng** [comment]
Please find the result in the link below. Thanks.
[result.txt](https://github.com/tensorflow/tensorflow/files/1895252/result.txt)

**Comment by mrry** [comment]
That seems to be a different failure (a Python `InvalidArgumentError` exception) than the one you described in the initial issue (a fatal error, leading to a process crash). Can you reproduce the process crash version with that environment variable set as well?

**Comment by hccheng** [comment]
Yes. It is kind of random actually; sometimes it's the InvalidArgumentError and sometimes it's the "ndims_byte() < MaxDimensions()" check fail. Please find the log below for the "ndims_byte() < MaxDimensions()":
[result_2.log](https://github.com/tensorflow/tensorflow/files/1895426/result_2.log)

**Comment by mrry** [comment]
It looks like the DynamicStitch op is giving an incorrect output in some cases. Can you replace the last part of the program with the following and let us know what the three values are when it crashes?

```python
op = tf.get_default_graph().get_operation_by_name("gradients/Sum_grad/DynamicStitch")
with tf.Session() as sess:
    for input_ in op.inputs:
        print(input_, sess.run(input_, feed_dict={x: Xdata, y: Ydata}))
    print(op.outputs[0], sess.run(op.outputs[0], feed_dict={x: Xdata, y: Ydata}))
    sess.run(tf.global_variables_initializer())
    sess.run(train_op, feed_dict={ x: Xdata, y: Ydata })
    print('done')
```

**Comment by hccheng** [comment]
Please find the logs below, thanks.
[Check failed.log](https://github.com/tensorflow/tensorflow/files/1898919/Check.failed.log)
[InvalidArgumentError.log](https://github.com/tensorflow/tensorflow/files/1898920/InvalidArgumentError.log)

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 89 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 105 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 119 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 134 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 149 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 164 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 179 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 194 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 210 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @mrry: It has been 225 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by mrry** [comment]
I'm afraid I haven't had time to reproduce this, and it's not an area that I work on, so I'm going to unassign myself (and hopefully someone else can pick up the issue, if it's still a problem).

**Comment by hgadig** [comment]
@mrry  Is this still applicable with the latest Tensorflow version ?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by jkulhanek** [comment]
I am having the same issue.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 29 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 44 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
We are closing this issue for now due to lack of activity. Please comment if this is still an issue for you. Thanks!
